contracts/strategies/stkCvxFXS/stkCvxFxsVault.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../GenericVault.sol";
+
+struct EarnedData {
+    address token;
+    uint256 amount;
+}
+
+interface stkCvxFxsStrategy {
+    function harvest(address _caller, uint256 _minAmountOut)
+        external
+        returns (uint256 harvested);
+}
+
+interface ICvxFxsStaking {
+    function claimableRewards(address _account)
+        external
+        view
+        returns (EarnedData[] memory userRewards);
+}
+
+contract stkCvxFxsVault is GenericUnionVault {
+    bool public isHarvestPermissioned = false;
+    mapping(address => bool) public authorizedHarvesters;
+
+    constructor(address _token) GenericUnionVault(_token) {}
+
+    /// @notice Sets whether only whitelisted addresses can harvest
+    /// @param _status Whether or not harvests are permissioned
+    function setHarvestPermissions(bool _status) external onlyOwner {
+        isHarvestPermissioned = _status;
+    }
+
+    /// @notice Adds or remove an address from the harvesters' whitelist
+    /// @param _harvester address of the authorized harvester
+    /// @param _authorized Whether to add or remove harvester
+    function updateAuthorizedHarvesters(address _harvester, bool _authorized)
+        external
+        onlyOwner
+    {
+        authorizedHarvesters[_harvester] = _authorized;
```

Comment:
Consider validating parameters (e.g. checking that `_harvester` is not the zero address) and emitting an event.